### PR TITLE
[Bug] Support all JA4_DB JAX fingerprint types and store them in DB. Closes #3541

### LINKED ISSUE #3541: refactor JA4_DB analyzer support all JAX fingerprint types and store them in database
## JA4_DB
- Currently only `JA4` traditional fingerprints are checked
- the comment mentions to support for all `JAX` types
- implement the support for all `JAX` types.
- store these fingerprints in the database instead of current `json` file storing structure

https://github.com/intelowlproject/IntelOwl/blob/5619186ef208d823c9de4754cdd2ca907733d750/api_app/analyzers_manager/observable_analyzers/ja4_db.py#L13-L33

### PR DESCRIPTION
**Pull request by chauhan-varun**
# Description

This PR fixes [#3541](https://github.com/intelowlproject/IntelOwl/issues/3541).

The existing `JA4_DB` analyzer was still tied to the old JSON-file cache and only handled the traditional JA4 lookup flow well. This change moves that storage into the database and expands the analyzer so it can work with all JAX fingerprint types currently returned by JA4DB, including `ja4s`, `ja4h`, `ja4x`, `ja4t`, `ja4ts`, and `ja4tscan`.

I added a new `Ja4DBEntry` model plus the related migration, updated the analyzer to populate and query that table, and removed the dependency on `ja4_db.json` for runtime lookups. I also added test coverage for the updater path, several JAX lookup paths, duplicate match handling, and update failure handling.

image
image

```
{
  "report": {
    "os": null,
    "notes": "",
    "device": null,
    "library": null,
    "verified": true,
    "application": "Nmap",
    "ja4_fingerprint": null,
    "ja4h_fingerprint": null,
    "ja4s_fingerprint": null,
    "ja4t_fingerprint": "1024_2_1460_00",
    "ja4x_fingerprint": null,
    "ja4ts_fingerprint": null,
    "observation_count": 1,
    "user_agent_string": null,
    "ja4tscan_fingerprint": null,
    "certificate_authority": null,
    "ja4_fingerprint_string": null
  },
  "data_model": null,
  "errors": [],
  "parameters": {}
}
```

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowlproject.github.io/docs/IntelOwl/contribute/) to this project
- [x] The pull request is for the branch `develop`
- [ ] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [ ] I strictly followed the documentation ["How to create a Plugin"](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-add-a-new-plugin)
    - [ ] [Usage](https://github.com/intelowlproject/docs/blob/main/docs/IntelOwl/usage.md) file was updated. A link to the PR to the [docs](https://github.com/intelowlproject/docs) repo has been added as a comment here.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/docs/blob/main/docs/IntelOwl/advanced_usage.md) was updated (in case the plugin provides additional optional configuration). A link to the PR to the [docs](https://github.com/intelowlproject/docs) repo has been added as a comment here.
    - [ ] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/api_app/analyzers_manager/test_classes.py).
    - [ ] If you created a new analyzer and it is free (does not require any API key), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-modify-a-plugin).
    - [ ] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowlproject.github.io/docs/IntelOwl/usage/#list-of-pre-built-playbooks).
    - [ ] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
    - [x] If the plugin interacts with an external service, I have created an attribute called precisely `url` that contains this information. This is required for Health Checks (HEAD HTTP requests).
    - [ ] If a new analyzer has beed added, I have created a unittest for it in the appropriate dir. I have also mocked all the external calls, so that no real calls are being made while testing.
    - [x] I have added that raw JSON sample to the `get_mocker_response()` method of the unittest class. This serves us to provide a valid sample for testing.
    - [ ] I have created the corresponding `DataModel` for the new analyzer following the [documentation](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-create-a-datamodel)
- [ ] I have inserted the copyright banner at the start of the file: ```# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl # See the file 'LICENSE' for copying permission.```
- [x] Please avoid adding new libraries as requirements whenever it is possible. Use new libraries only if strictly needed to solve the issue you are working for. In case of doubt, ask a maintainer permission to use a specific library.
- [ ] If external libraries/packages with restrictive licenses were added, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] Linters (`Ruff`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [x] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.
- [ ] After you had submitted the PR, if `DeepSource`, `Django Doctors` or other third-party linters have triggered any alerts during the CI checks, I have solved those alerts.

### CODE REVIEW

**Comment by chauhan-varun** [comment]
Hi @mlodic, this is ready for review Could you please take a look?

**Comment by mlodic** [comment]
it's not there are no screenshots or JSON results

**Comment by chauhan-varun** [comment]
I’ve updated the PR description and added both the screenshot and the raw JSON from a finished JA4_DB analysis, PTAL

**Comment by mlodic** [comment]
please show if the django admin section is populated properly.

File: api_app/analyzers_manager/observable_analyzers/ja4_db.py [hunk 1]
```diff
@@ -1,88 +1,77 @@
...
+            "details", flat=True
+        ):
+            serialized = json.dumps(details, sort_keys=True)
+            if serialized not in seen:
+                seen.add(serialized)
+                matches.append(details)
+        if len(matches) == 1:
+            return matches[0]
+        if matches:
+            return matches
```
Comment (mlodic) [inline api_app/analyzers_manager/observable_analyzers/ja4_db.py]: instead of doing this, please return "found": True and another key which is a list that contains the results.

Comment (chauhan-varun) [reply api_app/analyzers_manager/observable_analyzers/ja4_db.py]: Done! i updated it to return a consistent shape: `{"found": True, "results": [...]}` for matches, and adjusted the tests accordingly

**Comment by chauhan-varun** [comment]
> please show if the django admin section is populated properly.

image
